stacker/providers/aws/default.py

```diff
@@ -106,18 +111,54 @@ def cloudformation(self):
 
         return self._cloudformation
 
-    def get_stack(self, stack_name, **kwargs):
-        try:
-            return retry_on_throttling(
-                self.cloudformation.describe_stacks,
-                kwargs=dict(StackName=stack_name))['Stacks'][0]
-        except botocore.exceptions.ClientError as e:
-            if "does not exist" not in e.message:
-                raise
+    @property
+    def listener(self):
+        # deals w/ multiprocessing issues w/ sharing ssl conns
+        # see https://github.com/remind101/stacker/issues/196
+        pid = os.getpid()
+        if pid != self._pid or not self._listener:
+            session = get_session(self.region)
+            self._listener = CloudListener(
+                LISTENER_NAME, session=session)
+            self._listener.start()
+
+        return self._listener
+
+    def flush_events(self):
+        logger.info("Purging old queue")
+        while 1:
+            messages = self.listener.get_messages()
+            if len(messages) == 0:
+                break
+
+            for message in messages:
+                message.delete()
+        logger.info("Done purging queue")
+
+    def pre_run(self):
+        self.flush_events()
+
+    def parse_stack_events(self, tail):
+        messages = self.listener.get_messages()
+        for message in messages:
+            message_dict = message.__dict__
+            message_dict['fqn'] = message_dict['StackName']
+            self._stacks[message.StackName] = message_dict
+
+            if tail:
+                Provider._tail_print(message)
+
+        self.listener.delete_messages(messages)
+
+    def get_stack(self, stack_name, tail=False, **kwargs):
```

Comment:
This is kind of confusing - I know you were trying to avoid changing the actual methods that were called as much as possible, but I think this may just make it more confusing.  Also the fact we need to keep the dictionary of stacks seems like a complication that we might not necessarily need if this was refactored correctly.  Lets talk about this more.